src/angular-cesium/services/geo-utils/geo-utils.service.ts

```diff
@@ -5,22 +5,54 @@ import { Vec3 } from '../../models/vec3';
 
 @Injectable()
 export class GeoUtilsService {
-
-  static pointByLocationDistanceAndAzimuth(currentLocation: any, meterDistance: number, radianAzimuth: number, isInputCartesian = false) {
+  static pointByLocationDistanceAndAzimuth(currentLocation: any, meterDistance: number, radianAzimuth: number, deprecated?) {
     const distance = meterDistance / Cesium.Ellipsoid.WGS84.maximumRadius;
-    const curLat = isInputCartesian ? Cesium.Cartographic.fromCartesian(currentLocation).latitude : currentLocation.latitude;
-    const curLon = isInputCartesian ? Cesium.Cartographic.fromCartesian(currentLocation).longitude : currentLocation.longitude;
+    const cartographicLocation =
+      currentLocation instanceof Cesium.Cartesian3 ? Cesium.Cartographic.fromCartesian(currentLocation) : currentLocation;
+    const cartesianLocation =
+      currentLocation instanceof Cesium.Cartesian3
+        ? currentLocation
+        : Cesium.Cartesian3.fromRadians(currentLocation.longitude, currentLocation.latitude, currentLocation.height);
+
+    let resultPosition;
+    let resultDistance;
+    let counter = 0;
+    let distanceFactorRangeMax = 0.1;
+    let distanceFactorRangeMin = -0.1;
+    while (
+      counter === 0 ||
+      (counter < 16 && Math.max(resultDistance, meterDistance) / Math.min(resultDistance, meterDistance) > 1.000001)
+    ) {
+      const factor = distanceFactorRangeMin + (distanceFactorRangeMax - distanceFactorRangeMin) / 2;
+      resultPosition = GeoUtilsService._pointByLocationDistanceAndAzimuth(cartographicLocation, distance * (1 + factor), radianAzimuth);
+      resultDistance = this.distance(cartesianLocation, resultPosition);
+
+      if (resultDistance > meterDistance) {
+        distanceFactorRangeMax = distanceFactorRangeMin + (distanceFactorRangeMax - distanceFactorRangeMin) / 2;
+      } else {
+        distanceFactorRangeMin = distanceFactorRangeMin + (distanceFactorRangeMax - distanceFactorRangeMin) / 2;
```

Comment:
what about turf?

Reply:
It had the same issues as our original code